retrieval/target_provider.go

```diff
@@ -67,46 +78,79 @@ type sdTargetProvider struct {
 	refreshInterval time.Duration
 }
 
-// NewSdTargetProvider constructs a new sdTargetProvider for a job.
-func NewSdTargetProvider(job config.JobConfig) *sdTargetProvider {
-	i, err := utility.StringToDuration(job.GetSdRefreshInterval())
-	if err != nil {
-		panic(fmt.Sprintf("illegal refresh duration string %s: %s", job.GetSdRefreshInterval(), err))
-	}
+// NewSDTargetProvider constructs a new sdTargetProvider for a job.
+func NewSDTargetProvider(job config.JobConfig) TargetProvider {
 	return &sdTargetProvider{
 		job:             job,
-		refreshInterval: i,
+		refreshInterval: job.ServiceDiscoveryConfig().RefreshInterval(),
 	}
 }
 
-func (p *sdTargetProvider) Targets() ([]Target, error) {
-	var err error
+// Targets implements TargetPrvovider.
+func (p *sdTargetProvider) Targets() (targets []Target, err error) {
+	sd := p.job.ServiceDiscoveryConfig()
+	discoveryType := sd.GetType()
+
 	defer func() {
-		dnsSDLookupsCount.Inc()
```

Comment:
Maybe move this further down so we only increment if there was an actual lookup? This sort of skews the fail:success ratio.